### PR TITLE
feat: Support error code for single shot upload

### PR DESCRIPTION
**Pull request by Tulsishah**
Fixes https://github.com/googleapis/storage-testbench/issues/694

> It's a good idea to open an issue first for discussion.

- [x] Tests pass
- [ ] Appropriate changes to README are included in PR

### CODE REVIEW

File: testbench/rest_server.py [hunk 1]
```diff
@@ -979,6 +979,22 @@ def object_insert(bucket_name):
         blob, projection = gcs_type.object.Object.init_media(flask.request, bucket)
     elif upload_type == "multipart":
         blob, projection = gcs_type.object.Object.init_multipart(flask.request, bucket)
+        testbench.common.extract_instruction(request, context=None)
```
Comment (tritone) [inline testbench/rest_server.py:982]: This should work for both multipart and media; they are just diff flavors of single shot upload (though the Go SDK only uses multipart, but other SDKs might use media). See https://cloud.google.com/storage/docs/json_api/v1/objects/insert

File: testbench/rest_server.py [hunk 2]
```diff
@@ -979,6 +979,22 @@ def object_insert(bucket_name):
...
+        (
+            error_code,
+            after_bytes,
+            test_id,
+        ) = testbench.common.get_retry_uploads_error_after_bytes(db, request)
+        if error_code and len(blob.media) >= after_bytes:
+            if test_id:
+                db.dequeue_next_instruction(test_id, "storage.objects.insert")
+            testbench.error.generic(
+                "Fault injected during a resumable upload",
```
Comment (tritone) [inline testbench/rest_server.py:992]: Message should say single-shot upload here.

**Review by tritone — COMMENTED** [review]
Nice addition. We should look into making sure that this case is covered in the Go retry test scenarios as well.

File: testbench/rest_server.py (hunk 1 above)
Comment (Tulsishah) [reply testbench/rest_server.py:984]: Ohh. Thanks. Done. Supported for both.

File: testbench/rest_server.py (hunk 2 above)
Comment (Tulsishah) [reply testbench/rest_server.py]: Done

File: testbench/rest_server.py [hunk 3]
```diff
@@ -979,6 +979,8 @@ def object_insert(bucket_name):
         blob, projection = gcs_type.object.Object.init_media(flask.request, bucket)
     elif upload_type == "multipart":
         blob, projection = gcs_type.object.Object.init_multipart(flask.request, bucket)
+
+    if upload_type == "media" or upload_type == "multipart":
```
Comment (tritone) [inline testbench/rest_server.py:983]: I think you can remove this if -- the `if resumable` block above has a return so all remaining uploads are either media or multipart.

**Review by tritone — APPROVED** [review]
One more comment, otherwise LGTM

File: testbench/rest_server.py (hunk 3 above)
Comment (Tulsishah) [reply testbench/rest_server.py]: Done.

**Pull request merged**